client/client.go

```diff
@@ -12,63 +12,53 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Package client contains generic representations of clients connecting to different receivers
+// Package client contains generic representations of clients connecting to different receivers. Components,
+// such as processors or exporters, can make use of this information to make decisions related to
+// grouping of batches, tenancy, load balancing, tagging, among others.
 package client // import "go.opentelemetry.io/collector/client"
 
 import (
 	"context"
 	"net"
-	"net/http"
-
-	"google.golang.org/grpc/peer"
 )
 
 type ctxKey struct{}
 
-// Client represents a generic client that sends data to any receiver supported by the OT receiver
-type Client struct {
-	IP string
-}
+// Info contains data related to the clients connecting to receivers.
+type Info struct {
+	// IP for the client connecting to this collector. Available in a best-effort basis, and generally reliable
+	// for receivers making use of confighttp.ToServer and configgrpc.ToServerOption.
+	IP *net.IPAddr
 
-// NewContext takes an existing context and derives a new context with the client value stored on it
-func NewContext(ctx context.Context, c *Client) context.Context {
-	return context.WithValue(ctx, ctxKey{}, c)
+	// Auth information from the incoming request as provided by configauth.ServerAuthenticator implementations
+	// tied to the receiver for this connection.
+	Auth AuthData
 }
 
-// FromContext takes a context and returns a Client value from it, if present.
-func FromContext(ctx context.Context) (*Client, bool) {
-	c, ok := ctx.Value(ctxKey{}).(*Client)
-	return c, ok
-}
+// AuthData represents the authentication data as seen by authenticators tied to the receivers.
+type AuthData interface {
+	// Equal determines whether another authentication data is equal to this one.
+	// The actual semantics might be defined by the concrete implementations.
+	Equal(interface{}) bool
```

Comment:
Where is this used? Is this just test? Can we remove from the moment?

Reply:
Sure. This is going to be useful for the grouping/batching processors, but I can introduce it later.